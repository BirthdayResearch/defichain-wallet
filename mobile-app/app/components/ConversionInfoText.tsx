import { translate } from '@translations'
import { StyleProp, ViewStyle } from 'react-native'
import { InfoText } from './InfoText'

interface ConversionInfoTextProps {
  style?: StyleProp<ViewStyle>
}

export function ConversionInfoText (props: ConversionInfoTextProps): JSX.Element {
  return (
    <InfoText
      testID='conversion_info_text'
      text={translate('components/ConversionInfoText', 'Conversion will be required. Your passcode will be asked to authorize both transactions.')}
      style={props.style}
    />
  )
}
