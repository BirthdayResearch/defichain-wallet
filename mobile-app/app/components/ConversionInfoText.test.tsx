import { render } from '@testing-library/react-native'
import { ConversionInfoText } from './ConversionInfoText'

jest.mock('@shared-contexts/ThemeProvider')

describe('Conversion info text', () => {
  it('should match snapshot', async () => {
    const rendered = render(<ConversionInfoText />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
