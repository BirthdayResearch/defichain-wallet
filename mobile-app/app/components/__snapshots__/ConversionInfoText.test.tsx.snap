// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conversion info text should match snapshot 1`] = `
<View
  style={
    [
      [
        {},
        undefined,
      ],
      {},
    ]
  }
>
  <Text />
  <Text
    style={
      [
        {},
        [
          {},
          {},
        ],
      ]
    }
    testID="conversion_info_text"
    text="Conversion will be required. Your passcode will be asked to authorize both transactions."
  >
    Conversion will be required. Your passcode will be asked to authorize both transactions.
  </Text>
</View>
`;
